npipe_windows.go

```diff
@@ -328,21 +334,28 @@ func (l *PipeListener) AcceptPipe() (*PipeConn, error) {
 // Close stops listening on the address.
 // Already Accepted connections are not closed.
 func (l *PipeListener) Close() error {
+	l.listenerMutex.Lock()
```

Comment:
This won't work i'm afraid, if AcceptPipe has taken the lock, then Close will never be able to take the lock until accept returns, which it doesn't because close is the only thing (save an incoming connection) that can make accept return.